src/Packetizer/CCSDSPacketizer.c

```diff
@@ -0,0 +1,194 @@
+/**@file
+ * This file is part of the TASTE Runtime Common.
+ *
+ * @copyright 2022 N7 Space Sp. z o.o.
+ *
+ * TASTE Runtime Common was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CCSDSPacketizer.h"
+
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "IsoChecksum.h"
+#include "PacketizerInternal.h"
+#include "SpacePacketInternal.h"
+
+void
+CCSDS_Packetizer_init(Packetizer* const self)
+{
+    self->packetSequenceCount = 0;
+}
+
+size_t
+CCSDS_Packetizer_packetize(Packetizer* const self,
+                           const Packetizer_PacketType packetType,
+                           const uint16_t source,
+                           const uint16_t destination,
+                           uint8_t* const packetPointer,
+                           const size_t dataOffset,
+                           const size_t dataSize)
+{
+    // Unused in this implementation
+    (void)source;
+    (void)dataOffset;
```

Comment:
dataOffset is used in assert, right? Or does the compiler return a warning in release build?

Reply:
dataOffset is used in assert later on

Reply:
OK, do we need then the (void)dataOffset; ?

Reply:
Yes we do. If it is only in assert and not used later in code (referenced) then there is build error
![image](https://user-images.githubusercontent.com/121262861/215126423-ee2228fe-d5c2-4bbb-8022-e5286e835ad7.png)